File: 《亿级流量Jαva高并发与网络编程实战》-源码/demo/ch08/thrift/python/ThriftServer.py
#以python方法导入thrift相关类
from thrift.Thrift import TException
from thrift.protocol import TBinaryProtocol
from thrift.server import TServer
from thrift.transport import TSocket, TTransport

from StudentServiceImpl import StudentServiceImpl
from genpy.thrift.generatecode import StudentService

try:
   studentService  =  StudentServiceImpl()
   processor = StudentService.Processor(studentService)
   server = TSocket.TServerSocket("127.0.0.1",port=8888)
   #指定传输方式
   transportFactory =  TTransport.TFramedTransportFactory()
   #指定传输格式
   protocolFactory = TBinaryProtocol.TBinaryProtocolFactory()
   tServer = TServer.TSimpleServer(processor,server,transportFactory,protocolFactory)
   #启动服务
   tServer.serve()

except TException as e:
    print(e)

File: 《亿级流量Jαva高并发与网络编程实战》-源码/demo/ch08/thrift/nodejs/thriftServer.js
var thrift = require("thrift");
var StudentService = require('./StudentService');
//启动服务器，默认只支持TBufferedTransport和TBinaryProtocol
var server;
var transport = thrift.TFramedTransport;
var protocol = thrift.TBinaryProtocol;
var options = {transport: transport, protocol: protocol};

server = thrift.createServer(StudentService, {
    addStudent:function(name,age){
        console.log("--NodeJs服务端，模拟增加操作--") ;
        console.log(name+","+age+"增加成功！") ;
        return true ;
    },
    queryStudents:function () {
        console.log("--NodeJs服务端，模拟查询操作--") ;
        var students=[{name:"zs",age:23},{name:"ls",age:24}];
        console.log("--查询完毕--") ;
        return students ;
    }
},options);
server.listen(8888);